src/aws-adapter.js

```diff
@@ -179,7 +179,7 @@ export function createAdapter(opts = {}) {
         await con.log.flush();
       }
 
-      if (nonHttp) {
+      if (nonHttp || response.headers.get('force-non-http') === 'true') {
```

Comment:
I'm not very happy with the header name, as the `non-http` isn't really related to the response. one option would be to re-evaluate `context.invocation.event.requestContext`
and deleting it in the function. but this is also a bit ugly.

another option would be to return a status.code `0` or `999` (if fetch allows it) in order to mark it as non-http.

or use a header named `x-aws-raw-response`

Reply:
I like `x-aws-raw-response`.

Reply:
ok...I want to have @dominique-pfister input as well...

Reply:
> I like `x-aws-raw-response`.

same here 👍 

What I don't see at the moment, is why you'd want a HTTP request to return a non-HTTP response? Isn't it enough to directly invoke the Lambda and then benefit from `nonHttp` being true?